### PR TITLE
Catch installation errors and display them.

### LINKED ISSUE #193: Handle container install failure
The [`install_container()`](https://github.com/freedomofpress/dangerzone/blob/7bac3eb6b16c3ac2a2529105e2fa28f2bab3f7c3/dangerzone/global_common.py#L450) method returns false when it fails but none of its calls check the return value.

This should also be passed on to the user interface so the user can see the issue.

### PR DESCRIPTION
**Pull request by almet**
This catches the errors that can happen during installation, and during some other phases, and display them to the end user in the interface.

Fixes #193

### CODE REVIEW

**Comment by almet** [comment]
On MacOS, the tests are currently failing. I managed to reproduce on our mac mini. For a reason I still need to understand, the following command hangs indefinitely.

```bash
usr/local/bin/docker image list --format "{{.ID}}" dangerzone.rocks/dangerzone
```

And as a result the GUI continues showing that it's "installing" the image.

**Comment by almet** [comment]
Tests are now passing (without doing any changes). I'm not sure if it was due to an error with the runners? :thinking:
I re-triggered them and they seem to be green each time now.

@apyrgio I believe this is ready for review.

File: dangerzone/gui/main_window.py
Comment (apyrgio) [inline dangerzone/gui/main_window.py:502]: I would also add the following except clause:

```diff
+        except Exception as e:
+            log.error(str(e))
+            state = "not_running"
+            error = format_exception(e)
```

While toying with the PR (renaming the `docker.exe` file to something else), I managed to get an `OSError` exception here. It's best to catch everything, just to be on the safe side.

File: dangerzone/isolation_provider/container.py
Comment (apyrgio) [inline dangerzone/isolation_provider/container.py:226]: Don't know about you, but I think this method is very confusing. It checks if the container is installed, but also somehow has a side-effect (removing the container image if the image ID does not match). I think we can do better here, but that would be out of scope of this PR.

File: dangerzone/gui/main_window.py
Comment (almet) [reply dangerzone/gui/main_window.py:502]: Good catch! I've updated the code in 8da6d30af68979dc43239aa749071d967fa00441

File: dangerzone/isolation_provider/container.py
Comment (almet) [reply dangerzone/isolation_provider/container.py:226]: I agree: I was this 🤏🏼 close to propose a change there. Next time we touch here, I guess :+1:

**Review by apyrgio — APPROVED** [review]

**Comment by apyrgio** [comment]
Not much else to comment on, I think it's good to merge.

**Comment by almet** [comment]
Tests were intermittently failing on MacOS and Windows, due to the way we tested threads, which was a bit brittle. I changed the approach to flatten threads during the tests (replacing `.start()` with `.run()`) to ensure the logic is tested properly.

If that's okay for you @apyrgio, and that tests are green on all targets, I'll wait until #748 is merged, rebase and we should be good.

**Pull request merged**
